### PR TITLE
[DOCS] Python Exclusives overview

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Docs with overview of extensions provided by Python API.
 
### Tickets:
 - *79549*

### CODE REVIEW

File: docs/OV_Runtime_UG/Python_API_exclusives.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# OpenVINO™ Python API exclusives {#openvino_docs_OV_Runtime_UG_Python_API_exclusives}
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/Python_API_exclusives.md:1]: this page is not connected with other documentation. Need to put the `openvino_docs_OV_Runtime_UG_Python_API_exclusives` reference to any other document to make a connection

File: docs/snippets/ov_python_exclusives.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+print(model.outputs)
+
+print(compiled.inputs)
+print(compiled.outputs)
+#! [properties_example]
+
+#! [tensor_basics]
+data_float64 = np.ones(shape=(2,8))
+
+t = ov.Tensor(data_float64)
```
Comment (akuporos) [inline docs/snippets/ov_python_exclusives.py:30]: may be rename `t` to `tensor`?
I don't like that in public docs one-letter variable is used

File: docs/snippets/ov_python_exclusives.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+data = [np.array([i] * 8, dtype=np.float32) for i in range(jobs)]
+
+# Run all jobs
+for i in range(len(data)):
+    infer_queue.start_async({0: data[i], 1: data[i]})
+infer_queue.wait_all()
+#! [asyncinferqueue]
+
+#! [asyncinferqueue_access]
+results = infer_queue[3].get_output_tensor().data
```
Comment (AlexeyLebedev1) [inline docs/snippets/ov_python_exclusives.py:105]:
```suggestion
results = infer_queue[3].results
```

File: docs/snippets/ov_python_exclusives.py (hunk 2 above)
Comment (jiwaszki) [reply docs/snippets/ov_python_exclusives.py:30]: Done!

File: docs/snippets/ov_python_exclusives.py (hunk 3 above)
Comment (jiwaszki) [reply docs/snippets/ov_python_exclusives.py:105]: I will add new small snippet to cover some of many possibilities.

File: docs/OV_Runtime_UG/Python_API_exclusives.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+OpenVINO™ Runtime Python API is exposing additional features and helpers to elevate user experience. Main goal of Python API is to provide user-friendly and simple, yet powerful, tool for Python users.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply docs/OV_Runtime_UG/Python_API_exclusives.md:3]: Hmmm, I don't think it makes a big difference. @vyashina ?

**Review by AlexeyLebedev1 — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

File: docs/OV_Runtime_UG/Python_API_exclusives.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vyashina) [reply docs/OV_Runtime_UG/Python_API_exclusives.md:3]: "yet" doesn't fit the technical documentation.
it'd be better to use "but" or "still".
But it's not a critical issue for sure.

File: docs/OV_Runtime_UG/Python_API_exclusives.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [properties_example]
+
+@endsphinxdirective
+
+Please refer to Python API documentation on which helper functions or properties are available for different classes.
```
Comment (vyashina) [inline docs/OV_Runtime_UG/Python_API_exclusives.md:29]: Refer to Python API documentation on which helper functions or properties are available for different classes.

File: docs/OV_Runtime_UG/Python_API_exclusives.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+## Working with Tensor
+
+Python API allows users to pass data as tensors. `Tensor` object holds a copy of the data from the given array. `dtype` of numpy arrays is converted to OpenVINO™ types automatically.
```
Comment (vyashina) [inline docs/OV_Runtime_UG/Python_API_exclusives.md:33]: Python API allows you to pass data as tensors OR
Python API allows passing data as tensors.

File: docs/OV_Runtime_UG/Python_API_exclusives.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [asyncinferqueue_access]
+
+@endsphinxdirective
+
+#### Setting callbacks
+
+Another feature of `AsyncInferQueue` is ability of setting callbacks. When callback is set, any job that ends inference, calls upon Python function. Callback function must have two arguments. First is the request that calls the callback, it provides `InferRequest` API. Second one being called "userdata", passed values can be of any Python type and later used inside callback function.
```
Comment (vyashina) [inline docs/OV_Runtime_UG/Python_API_exclusives.md:133]: Second one,  called "userdata", provides (?) values, which can be of any Python type and later used inside callback function.

Comment (jiwaszki) [reply docs/OV_Runtime_UG/Python_API_exclusives.md]: Done

File: docs/OV_Runtime_UG/Python_API_exclusives.md (hunk 6 above)
Comment (jiwaszki) [reply docs/OV_Runtime_UG/Python_API_exclusives.md]: Done

File: docs/OV_Runtime_UG/Python_API_exclusives.md (hunk 5 above)
Comment (jiwaszki) [reply docs/OV_Runtime_UG/Python_API_exclusives.md]: Done

**Pull request merged**